lib/rucio/client/uploadclient.py

```diff
@@ -634,7 +677,7 @@ def _upload_item(self, rse_settings, rse_attributes, lfn, source_dir=None, domai
                 pfn = self.client.get_signed_url(rse_settings['rse'], sign_service, 'write', pfn)
 
         # Create a name of tmp file if renaming operation is supported
-        pfn_tmp = '%s.rucio.upload' % pfn if protocol_write.renaming else pfn
+        pfn_tmp = cast(str, '%s.rucio.upload' % pfn if protocol_write.renaming else pfn)
```

Comment:
Why is cast necessary here? Are there cases where pfn isn't already a string?

Reply:
The static type checker thinks that `pfn` could be `None`, which occurs in the event that:
- the `Exception` occurs on line 665
- `force_pfn` is `None`
- `if sign_service` is not `True`

I don't know if this edge case can realistically occur. I am using `cast` there because `pfn_tmp` is used in a few places afterwards that expect a `str` - if I were to `# type: ignore` then I would have to ignore multiple lines instead of just casting on one